crates/bevy_diagnostic/src/diagnostic.rs

```diff
@@ -28,6 +28,30 @@ pub struct DiagnosticMeasurement {
     pub value: f64,
 }
 
+/// Enum for display percision. Choose `IntegerValue` for discrete values and `DecimalValue(num_of_decimals)` to specify the number of decimals
+#[derive(Debug)]
+pub enum DisplayPercision {
+    IntegerValue,
+    DecimalValue(usize),
+}
```

Comment:
Why have an enum for this?

Running the code in the rust playground:

```rust
fn main() {
    let x = 5.678912345;
    dbg!(format!("{x:.6}"));
    dbg!(format!("{x:.0}"));
}
```

This prints:

```
[src/main.rs:3] format!("{x:.6}") = "5.678912"
[src/main.rs:4] format!("{x:.0}") = "6"
```

I don't think an enum is necessary? You can just use `0` to print an integer.

Reply:
I just added the enum. Just using a usize is how I did it to begin with.

Reply:
Hmm. @alice-i-cecile I have a hard time understanding why turning this into an enum is better. Can you clarify why?

Reply:
Certain diagnostics will *always* be integer-valued: users should not be able to configure the number of decimals for them. Other diagnostics are float-like, and users may want to round to the nearest whole number.

To me, these cases are distinct (even though the same number of decimal places are shown). Representing that in the type system means that we'll be able to expose the correct controls to users later.

Reply:
Do `IntegerValue` and `DecimalValue(0)` have different behavior?

Reply:
Not in the current version of this PR. I'll play with this by next Monday, and try to get a full design here as a PR to @zeroacez branch :)